image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
 
-This allows consumers of your image to customize behavior (e.g. database settings, passwords, performance tuning) without having to introduce a new layer on top of your image.  Instead, they can simply define environment variable values when defining a pod, and change those settings without rebuilding the image.
+In addition, tuning should be done by inspecting the [sysitem]#cgroups# settings for the container. This allows the image to tune itself to the available memory, CPU, and other resources. For example, Java-based images should tune their heap based on the [sysitem]#cgroup# maximum memory parameter to ensure they do not exceed the limits and get an out-of-memory error.
 
-For extremely complex scenarios, configuration can also be supplied via volumes which are expected to be mounted into the container at runtime.  If you take this path, your image should provide clear error messages on startup when the necessary volume or configuration is not present.
+See the following references for more on how to manage [sysitem]#cgroup# quotas in Docker containers:
 
-This topic ties in with using services for inter-image communication in that configuration like datasources should be defined in terms of environment variables which provide the service endpoint information.  In this way, an application can dynamically consume a datasource service that is defined in the OpenShift environment without modifying the application image.
+- https://goldmann.pl/blog/2014/09/11/resource-management-in-docker
+- https://docs.docker.com/articles/runmetrics
+- http://fabiokung.com/2014/03/13/memory-inside-linux-containers
 
-In addition, tuning should be done by inspecting the cgroups settings for the container.  This will allow the image to tune itself to the available memory, CPU, etc.  For example java based images should tune their heap based on the cgroup maximum memory parameter to ensure they do not exceed the limits and get an out of memory error.
+*Clustering*
 
-Here are some useful references on how to manage cgroup quotas in Docker containers:
+Give careful thought to what it means to run multiple instances of your image. In the simplest case, the load balancing function of a service handles routing traffic to all instances of your image.  However, many frameworks need to share information in order to perform leader election or failover state, for example in session replication.
 
-* https://goldmann.pl/blog/2014/09/11/resource-management-in-docker/
-* https://docs.docker.com/articles/runmetrics/
-* http://fabiokung.com/2014/03/13/memory-inside-linux-containers/
+Consider how your instances accomplish this when running in OpenShift. Pods can communicate directly with each other, however their IP addresses change anytime the pod stops, starts, or moves, so your clustering scheme must be very dynamic.
 
```

Comment:
[Question]
"Accomplish" what?

[Suggestion]
Although pods can communicate directly with each other, their IP addresses change anytime the pod starts, stops, or is moved. Therefore, it is important for your clustering scheme to be dynamic.